File: python/merge/Merger_ps_pdf.py
from aspose.pdf import (
    SaveFormat,
    PsLoadOptions,
    Document
)


def ps_to_pdf(self):
    pathSource1 = "../../TestData/test.ps"
    pathSource2 = "../../TestData/Second/test.ps"

    # PostScript files can be parsed and loaded as Aspose Document
    firstDoc = Document(pathSource1, PsLoadOptions)
    secondDoc = Document(pathSource2, PsLoadOptions)

    # create empty pdf document
    outputDoc = Document()

    # set less memory usage with unload instead of fast performance
    outputDoc.enable_object_unload = True

    for page in firstDoc.Pages:
        # add page from one document to another directly
        outputDoc.Pages.Add(page)

    for page in secondDoc.Pages:
        # add page from one document to another directly
        outputDoc.Pages.Add(page)

    # save result pdf to file
    outputDoc.Save("test.pdf", SaveFormat.Pdf)

File: cpp/merge/Merger_ps_pdf.cpp
#include "Aspose.PDF.Cpp/Document.h"
#include "Aspose.PDF.Cpp/PsLoadOptions.h"
#include "Aspose.PDF.Cpp/Page.h"
#include "Aspose.PDF.Cpp/PageCollection.h"
#include "Aspose.PDF.Cpp/SaveFormat.h"
using namespace System;
using namespace Aspose::Pdf;

void ps_to_pdf()
{
    String pathSource1 = u"../../TestData/test.ps";
    String pathSource2 = u"../../TestData/Second/test.ps";

    // PostScript files can be parsed and loaded as Aspose Document
    auto firstDoc = MakeObject<Document>(pathSource1, MakeObject<PsLoadOptions>());
    auto secondDoc = MakeObject<Document>(pathSource2, MakeObject<PsLoadOptions>());

    // create empty pdf document
    auto outputDoc = MakeObject<Document>();

    // set less memory usage with unload instead of fast performance
    outputDoc->EnableObjectUnload = true;

    for (int i = 1; i < firstDoc->get_Pages()->get_Count(); i++)
    {
        auto page = firstDoc->get_Pages()->idx_get(i);
        // add page from one document to another directly
        outputDoc->get_Pages()->Add(page);
    }

    for (int i = 1; i < secondDoc->get_Pages()->get_Count(); i++)
    {
        auto page = secondDoc->get_Pages()->idx_get(i);
        // add page from one document to another directly
        outputDoc->get_Pages()->Add(page);
    }

    // save result pdf to file
    outputDoc->Save(u"test.pdf", SaveFormat::Pdf);
}
